app/server/lib/ExternalStorage.ts

```diff
@@ -226,13 +228,25 @@ export class ChecksummedExternalStorage implements ExternalStorage {
           const expectedChecksum = await this._options.sharedHash.load(fromKey);
           // Let null docMD5s pass.  Otherwise we get stuck if redis is cleared.
           // Otherwise, make sure what we've got matches what we expect to get.
-          // S3 is eventually consistent - if you overwrite an object in it, and then read from it,
-          // you may get an old version for some time.
+          // AWS S3 was eventually consistent, but now has stronger guarantees:
+          // https://aws.amazon.com/blogs/aws/amazon-s3-update-strong-read-after-write-consistency/
+          //
+          // Previous to this change, if you overwrote an object in it,
+          // and then read from it, you may have got an old version for some time.
+          // We are confident this should not be the case anymore, though this has to be studied carefully.
           // If a snapshotId was specified, we can skip this check.
           if (expectedChecksum && expectedChecksum !== checksum) {
-            log.error("ext %s download: data for %s has wrong checksum: %s (expected %s)",
-                      this.label, fromKey, checksum, expectedChecksum);
-            return undefined;
+            const message = `ext ${this.label} download: data for ${fromKey} has wrong checksum:` +
+              ` ${checksum} (expected ${expectedChecksum})`;
+
+            // Only warn if GRIST_SKIP_REDIS_CHECKSUM_MISMATCH is set. This flag is experimental
```

Comment:
I realized this description is ambiguous if you don't happen to know what the previous behavior was. Can you do "If GRIST_SKIP_REDIS_CHECKSUM_MISMATCH is set, issue a warning only and continue, rather than issuing an error and failing."

Also, can you add the new env variable to README.md, even though it may be only temporary? Temporary variables has a habit of sticking around longer than you might expect :-)